indra/llcharacter/llik.cpp

```diff
@@ -1407,6 +1426,46 @@ void LLIK::Joint::setLocalRot(const LLQuaternion& new_local_rot)
     }
 }
 
+// only call this if you know what you're doing
+// this should only be called once before starting IK algorithm iterations
+void LLIK::Joint::setLocalScale(const LLVector3& scale)
+{
+    // compute final scale adustment to applly to mLocalPos and mBone
+    constexpr F32 MIN_INVERTABLE_SCALE = 1.0e-15f;
+    LLVector3 re_scale;
+    for (S32 i = 0; i < 3; ++i)
+    {
+        // verify mLocalScale component to avoid introducing NaN
+        re_scale[i] = (mLocalScale[i] > MIN_INVERTABLE_SCALE) ?  scale[i] / mLocalScale[i] : 0.0f;
+    }
+    // We remember the final scale adjustment for later...
+    mLocalScale = scale;
+    // ...and apply it immediately onto mLocalPos and mBone.
+    mBone.scaleVec(re_scale);
+    mLocalPos.scaleVec(re_scale);
+    mLocalPosLength = mLocalPos.length();
+}
+
+// returns local_pos with any non-uniform scale from the "info" removed.
+LLVector3 LLIK::Joint::getPreScaledLocalPos() const
+{
+    LLVector3 pos = mLocalPos;
+    // We inverse-scale mLocalPos because we already applied the info's scale
+    // to mLocalPos so we could perform IK without constantly recomputing it,
+    // and now we're being asked for mLocalPos in the info's pre-scaled frame.
+
+    // Note: mInfoPtr will not be nullptr as per assert() in LLIK::Joint::ctor
+    LLVector3 inv_scale = mInfoPtr->getScale();
+    constexpr F32 MIN_INVERTABLE_SCALE = 1.0e-15f;
+    for (S32 i = 0; i < 3; ++i)
+    {
+        // verify scale component to avoid introducing NaN
+        inv_scale[i] = (inv_scale[i] > MIN_INVERTABLE_SCALE) ?  1.0f / inv_scale[i] : 0.0f;
+    }
+    pos.scaleVec(inv_scale);
+    return pos;
+}
```

Comment:
At harvest time we want the local position in the **LLJoint** frame, so we need to back out any **LLJoint** scaling we applied before IK calculations were started.